metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -126,13 +125,16 @@ class MetalsLspService(
     shellRunner: ShellRunner,
     timerProvider: TimerProvider,
     initTreeView: () => Unit,
-    val folder: AbsolutePath,
+    folder: AbsolutePath,
     folderVisibleName: Option[String],
     headDoctor: HeadDoctor,
 ) extends Cancelable
-    with TextDocumentService {
+    with TextDocumentService
+    with FolderLike {
```

Comment:
Maybe `extends Folder(folder, folderVisibleName,  forceIsScalaProject = true) with Cancellable ...` and you wouldn't need the additional FolderLike trait, which is not that useful as it's not sealed so it's one additional level of indirection.